### PR TITLE
Market maker bot work in progress

### PR DESCRIPTION
**Pull request by andrlime**
Some questions:
1. What is the return type of take_action and how do I compute that?
2. What do I do with the spreads?
3. Are the types of the variables correct?
4. What do I need to do in bot.hpp?

### CODE REVIEW

**Comment by echavemann** [comment]
Types r fine

**Comment by echavemann** [comment]
Wdym what do with spreads

**Comment by stevenewald** [comment]
> Types r fine

they aint

File: exchange/src/bots/mm.hpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
+
+#include <vector>
+
+namespace nutc {
+namespace bots {
+
+class MarketMakerBot {
+public:
+    float long_positions;
+    float short_positions;
```
Comment (egelja) [inline exchange/src/bots/mm.hpp:10]: this should be an array no?

File: exchange/src/bots/mm.hpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    float long_capital;
+    float short_capital;
```
Comment (egelja) [inline exchange/src/bots/mm.hpp:13]: initialize to 0

File: exchange/src/bots/mm.hpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    float capital_limit;
```
Comment (egelja) [inline exchange/src/bots/mm.hpp:15]: should be ctor argument

File: exchange/src/bots/mm.hpp [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    // TODO: fix input
+    explicit inline MarketMakerBot() {}
+    // TODO: implement
+    std::vector<float> take_action(float new_theo);
```
Comment (egelja) [inline exchange/src/bots/mm.hpp:20]: wat is `new_theo`?

File: exchange/src/bots/mm.hpp [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+private:
+    inline float compute_net_exposure() {
+        return (long_positions - short_positions);
+    }
+    inline float compute_capital_util() {
+        return (long_capital + short_capital)/capital_limit;
+    }
+    inline float compute_capital_tolerance() {
+        return (-1 * compute_net_exposure()) * (1 + (1 - compute_capital_util()) * capital_limit/2);
+    }
```
Comment (egelja) [inline exchange/src/bots/mm.hpp:30]: postfix with `_` for private methods

also you don't need inline if its in a class

File: exchange/src/bots/mm.hpp [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+    // TODO: fix input
+    explicit inline MarketMakerBot() {}
+    // TODO: implement
+    std::vector<float> take_action(float new_theo);
+private:
+    inline float compute_net_exposure() {
+        return (long_positions - short_positions);
+    }
+    inline float compute_capital_util() {
```
Comment (egelja) [inline exchange/src/bots/mm.hpp:25]: util?

**Review by egelja — CHANGES_REQUESTED** [review]

**Comment by egelja** [comment]
also wait why are all fields public?

**Pull request merged**